src/outputs/output_utils.cpp

```diff
@@ -100,6 +100,71 @@ AllSwarmInfo::AllSwarmInfo(BlockList_t &block_list,
   }
 }
 
+// Tools that can be shared accross Output types
+
+std::vector<Real> ComputeXminBlocks(Mesh *pm) {
+  return FlattenBlockInfo<Real>(pm, 3,
+                                [=](MeshBlock *pmb, std::vector<Real> &data, int &i) {
+                                  auto xmin = pmb->coords.GetXmin();
+                                  data[i++] = xmin[0];
+                                  if (pm->ndim > 1) {
+                                    data[i++] = xmin[1];
+                                  }
+                                  if (pm->ndim > 2) {
+                                    data[i++] = xmin[2];
+                                  }
+                                });
```

Comment:
Is there a potential danger for misuse by accident here for outputs with ndim != 3 because the returned vector's size will always be `3*num_blocks_local` while there might only be valid data in one or two thirds.

Reply:
Hmm... yes there probably is. I believe this refactor maintains the behavior of the original, but let me think about edge cases here... Maybe the original code was a little dangerous.

Reply:
I think there's no danger in changing `3` to `ndim` here so I'm just going to make that change.